### DISCUSSION TITLE
Did TW pass any CSS optimizations?

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#6441] [General]
Running tiddlywiki.com through https://purifycss.online/# gives some very minor improvement proposals. I suspect the "minuteness" of the proposals may just be that such online services can't quite handle something as complex as TW. Regardless, looking at the resulting "diff comparison" it proposes to change rgb colors into hex, e.g `rgb(255,75,75)` into `#ff4b4b`. That's a few bytes but maybe there are several such instances, and maybe - and this is my main point with this post - maybe the CSS has never been run through one of these CSS optimization tools. Has it? Is it worth the bother?

**Comment by Jermolene** [comment]
Hi @twMat that tool only reads the boot stylesheet embedded in the TiddlyWiki HTML file. It doesn't see the stylesheets tagged `$:/tags/StyleSheet` that are created manually.

> maybe the CSS has never been run through one of these CSS optimization tools. Has it? Is it worth the bother?

I've not used any tools myself. I'd guess that tools that work as bookmarklets or browser extensions would be more likely to be able to process TiddlyWiki's dynamically created CSS.

**Comment by pmario** [comment]
Such tools typically compare the CSS with the page, that is currently shown. So they only see if there are CSS definitions, that are not used at the moment. So they will probably show, that 70% of the CSS isn't used, because they are not active, when you make the test. 

TW is dynamic and we probably need all the definitions that are part of the style sheets. But we don't need it at the same time.
